### ISSUE TITLE
dealing with multiple encodings in figures

### ISSUE THREAD

**Issue by amueller** [jupyter/nbdime#224]
I'm trying to diff a notebook where figures are encoded in multiple ways, like so:

```python
from IPython.display import set_matplotlib_formats
import numpy as np
import matplotlib.pyplot as plt


set_matplotlib_formats('png', 'svg')
#set_matplotlib_formats('pdf', 'png')

%matplotlib inline
plt.imshow(np.eye(5))
```

Using either ``svg`` or ``pdf``, I get base64 encoded images in the nbdiff-web view when diffing against an empty notebook.

Actually, if I set only ``svg``, I get the xml representation. If I set ``'png', 'svg'`` I get a base64 encoding of the png, I think, and don't see a png image.

**Comment by amueller** [comment]
Update: doing pdf alone actually embeds a link to a pdf, which I I classify as "working".
So it's really about not being able to deal with multiple encodings in figures.

There's also the separate issue of not doing SVG rendering.

**Comment by vidartf** [comment]
Thanks for this! The issue is that it doesn't try to render the output if not *all* the MIME types of the output are safe or sanitizable. This behavior should be changed to checking the MIME types that are the preferred candidate.

**Comment by vidartf** [comment]
A related question is: If the MIME type that the notebook app would prefer is unsafe, should we show its diff as text, or should we instead try to render the next preferred MIME type?

**Comment by amueller** [comment]
I'm not sure I understand about the MIME types. Individually pdf and png are rendered, but not if both are present.

**Comment by vidartf** [comment]
Don't worry about it, I added the comment above more as a note-to-self for tracking the issue better :)

**Comment by vidartf** [comment]
I submitted a fix in #228.
